olxcleaner/loader/xml.py

```diff
@@ -109,14 +109,21 @@ def read_course(edxobj, node, directory, filename, errorstore, htmlfiles, pointe
                     empty = False
 
     # Check for a pointer tag
+    node_filename = os.path.splitext(os.path.split(filename)[-1])[0]
     if empty and edxobj.is_pointer(node.attrib):
-
         if pointer:
-            # The target of a pointer tag cannot be a pointer itself
-            errorstore.add_error(SelfPointer(filename,
-                                             edxobj=edxobj))
-            edxobj.broken = True
-            return
+            if node.attrib.get('url_name') == node_filename:
+                # The target of a pointer tag cannot be a pointer itself
+                errorstore.add_error(SelfPointer(filename,
+                                                 edxobj=edxobj))
+                edxobj.broken = True
```

Comment:
It's important to know when to mark an object `broken` and return. I think @jolyonb can add some thoughts here.

Reply:
Any object that should not be subject to further examination in later stages of processing should be marked as broken.

Reply:
(It's mostly only used in loading, to signify that the object was unable to load in a valid manner.)